lib/msRest.ts

```diff
@@ -4,7 +4,7 @@
 export { WebResource, HttpRequestBody, RequestPrepareOptions, HttpMethods, ParameterValue, RequestOptionsBase, TransferProgressEvent, AbortSignalLike } from "./webResource";
 export { DefaultHttpClient } from "./defaultHttpClient";
 export { HttpClient } from "./httpClient";
-export { HttpHeaders } from "./httpHeaders";
+export { IHttpHeaders, HttpHeaders } from "./httpHeaders";
```

Comment:
I'm torn on whether this should actually be exported from the module's public interface as it seems unnecessary.  Thoughts?

Reply:
I think, in TypeScript it is okay to have an interface and a class with the same name. So how about naming the interface HttpHeaders as well.

Reply:
That's a great idea!  Initial testing shows it might work, stay tuned...

Reply:
Another thought, What if we name the interface HttpHeaders and the class as `HttpHeadersImpl`. This shouldn't be a breaking change in TypeScript since, TS focuses on the structure and not the name. AFAIK, you have simply added an interface that declares the methods and properties of what was the `HttpHeaders` **class**. If that is true, then renaming the class to `HttpHeadersImpl` should work. `HttpHeaders` is already exported. So everything should work business as usual.

Reply:
Looks like TypeScript can't disambiguate between the class an interface well enough when they have the same name.  For now I'll stick with `IHttpHeaders` unless someone has a better idea for its name

Reply:
(whoops, wrote my last comment without reloading to see @amarzavery's latest)

Already chatted with Amar offline about this, I think it'll be a breaking change to rename `HttpHeaders` to `HttpHeadersImpl` if anyone using `ms-rest-js` is creating a `HttpHeaders` instance.